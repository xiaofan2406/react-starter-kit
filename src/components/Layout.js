import React from 'react';
import injectSheet, { fontFamily, fontSize } from 'styles';
import 'styles/reset.css';
import 'styles/animation.css';

import Header from './Header';


const styles = {
  layout: {
    fontFamily,
    fontSize
  },
  main: {
    padding: '1em'
  }
};

/* Use functions rather than constant elements for better debugging */
function Layout({ sheet: { classes }, children }) {
  return (
    <div className={classes.layout}>
      <Header />
      <div className={classes.main}>
        {children}
      </div>
    </div>
  );
}

Layout.propTypes = {
  sheet: React.PropTypes.object.isRequired,
  children: React.PropTypes.node.isRequired
};


export default injectSheet(styles)(Layout);
