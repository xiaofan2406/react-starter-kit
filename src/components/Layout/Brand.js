import React, { memo } from 'react';
import { APP_TITLE } from '../../utils/constants';
import logo from '../../assets/logo.svg';

const Brand = () => (
  <div className="flex items-center p-2">
    <img src={logo} alt="logo" className="spin h-10" />
    <span className="fade-in text-lg">{APP_TITLE}</span>
  </div>
);

export default memo(Brand);
