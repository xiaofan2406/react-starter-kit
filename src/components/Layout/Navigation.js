import React from 'react';
import PropTypes from 'prop-types';
import withCss from 'react-jss';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'configs';
import { theme, variables } from 'styles';

const css = {
  Navigation: {},
  link: {
    color: theme.inverseColor,
    textDecoration: 'none',
    padding: '0 0.5em',
    display: 'inline-block',
    lineHeight: '42px',
    height: variables.Layout.headerHeight,
    '&:hover': {
      backgroundColor: theme.bgAccentColor
    }
  },
  linkActive: {
    borderBottom: `2px solid ${theme.primaryColor}`
  }
};

function Navigation({ classes }) {
  return (
    <div className={classes.Navigation}>
      {Object.values(ROUTES).map(route =>
        <NavLink
          className={classes.link}
          activeClassName={classes.linkActive}
          key={route.path}
          exact={route.exact}
          to={route.path}
        >
          {route.name}
        </NavLink>
      )}
    </div>
  );
}

Navigation.propTypes = {
  classes: PropTypes.object.isRequired
};

export { Navigation as Component };

export default withCss(css)(Navigation);
