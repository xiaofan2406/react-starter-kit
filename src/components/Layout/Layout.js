import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'react-emotion';
import { theme, spacing } from 'styles';
import Brand from './Brand';

const cssLayout = css`
  font-family: ${theme.fontFamily};
  font-size: ${theme.fontSize}px;
  color: ${theme.textColor};
  & p {
    margin: 0px 0px 1em 0px;
  }
  & > .main {
    width: 780px;
    padding: 0px ${spacing.internal}px;
    margin: ${spacing.break}px auto;
  }
`;

const Layout = ({ children }) => (
  <div className={cssLayout}>
    <Brand />
    <div className="main">{children}</div>
  </div>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export { Layout as Component };

export default Layout;
