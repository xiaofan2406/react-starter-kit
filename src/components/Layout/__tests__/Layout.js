import React from 'react';
import { shallow } from 'enzyme';

import { Component } from '../Layout';

it('should match snapshot', () => {
  const props = { classes: {}, children: '' };
  const component = shallow(<Component {...props} />);

  expect(component).toMatchSnapshot();
});

it('should render children', () => {
  const Child = () => <div>child</div>;
  const props = { classes: {}, children: <Child /> };
  const component = shallow(<Component {...props} />);

  expect(component).toMatchSnapshot();
  expect(component.find(Child).length).toBe(1);
});
