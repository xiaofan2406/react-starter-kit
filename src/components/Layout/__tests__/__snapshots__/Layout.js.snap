// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="css-k0jtqr"
>
  <div
    className="header"
  >
    <Brand />
    <Navigation />
  </div>
  <div
    className="main"
  />
</div>
`;

exports[`should render children 1`] = `
<div
  className="css-k0jtqr"
>
  <div
    className="header"
  >
    <Brand />
    <Navigation />
  </div>
  <div
    className="main"
  >
    <Child />
  </div>
</div>
`;
